client/matching/client.go

```diff
@@ -155,6 +159,15 @@ func (c *clientImpl) PollWorkflowTaskQueue(
 }
 
 func (c *clientImpl) QueryWorkflow(ctx context.Context, request *matchingservice.QueryWorkflowRequest, opts ...grpc.CallOption) (*matchingservice.QueryWorkflowResponse, error) {
+	// use shallow copy since QueryRequest may contain a large payload
```

Comment:
nit: It could be nice if all of these APIs behaved the same, and we didn't need to think about whether to shallow copy or deep copy the request. We could define a helper "copy" function that basically does a shallow copy of everything except the task queue, and then use it everywhere. Since the task queue is the main thing we need to protect from mutation here.

Reply:
I agree, but they're all separate types so we'd need a bunch of functions. I think we should just generate a shallow copy helper in our proto helpers and use that for all these (and many other places I've wanted a shallow copy)